store/sqlstore/channel_store.go

```diff
@@ -510,64 +510,62 @@ func (s SqlChannelStore) CreateDirectChannel(userId string, otherUserId string)
+		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.open_transaction.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+	defer finalizeTransaction(transaction)
+
+	directchannel.TeamId = ""
+	// After updating saveChannelT() should be:
+	// newChannel, apperr := s.saveChannelT(transaction, directchannel, 0)
+	channelResult := s.saveChannelT(transaction, directchannel, 0)
+	var newChannel *model.Channel
+	if channelResult.Data != nil {
+		newChannel = channelResult.Data.(*model.Channel)
+	}
+	apperr := channelResult.Err
 
-		directchannel.TeamId = ""
-		channelResult := s.saveChannelT(transaction, directchannel, 0)
+	if apperr != nil {
+		return newChannel, apperr
+	}
 
-		if channelResult.Err != nil {
-			result.Err = channelResult.Err
-			result.Data = channelResult.Data
-			return
-		}
+	// Members need new channel ID
+	member1.ChannelId = newChannel.Id
+	member2.ChannelId = newChannel.Id
 
-		newChannel := channelResult.Data.(*model.Channel)
-		// Members need new channel ID
-		member1.ChannelId = newChannel.Id
-		member2.ChannelId = newChannel.Id
+	member1Result := s.saveMemberT(transaction, member1, newChannel)
+	member2Result := member1Result
+	if member1.UserId != member2.UserId {
+		member2Result = s.saveMemberT(transaction, member2, newChannel)
+	}
 
-		member1Result := s.saveMemberT(transaction, member1, newChannel)
-		member2Result := member1Result
-		if member1.UserId != member2.UserId {
-			member2Result = s.saveMemberT(transaction, member2, newChannel)
+	if member1Result.Err != nil || member2Result.Err != nil {
+		details := ""
+		if member1Result.Err != nil {
+			details += "Member1Err: " + member1Result.Err.Message
 		}
-
-		if member1Result.Err != nil || member2Result.Err != nil {
-			details := ""
-			if member1Result.Err != nil {
-				details += "Member1Err: " + member1Result.Err.Message
-			}
-			if member2Result.Err != nil {
-				details += "Member2Err: " + member2Result.Err.Message
-			}
-			result.Err = model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.add_members.app_error", nil, details, http.StatusInternalServerError)
-			return
+		if member2Result.Err != nil {
+			details += "Member2Err: " + member2Result.Err.Message
 		}
+		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.add_members.app_error", nil, details, http.StatusInternalServerError)
+	}
 
-		if err := transaction.Commit(); err != nil {
-			result.Err = model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.commit.app_error", nil, err.Error(), http.StatusInternalServerError)
-			return
-		}
+	if err := transaction.Commit(); err != nil {
+		return nil, model.NewAppError("SqlChannelStore.SaveDirectChannel", "store.sql_channel.save_direct_channel.commit.app_error", nil, err.Error(), http.StatusInternalServerError)
+	}
+
+	return newChannel, apperr
```

Comment:
Here `apperr` must be nil, so, is way clearer to just write `return newChannel, nil`